### PR TITLE
Document how to deploy Control Center with a custom ingress

### PR DESCRIPTION
**Pull request by tamasmak**
This documents how to deploy Control Center with a custom ingress configuration and separates the instructions into development and production environment parts.

Closes [vaadin/control-center#592](https://github.com/vaadin/control-center/issues/592)

### CODE REVIEW

File: articles/control-center/getting-started/index.adoc [hunk 1]
```diff
@@ -37,8 +39,48 @@ helm install control-center oci://docker.io/vaadin/control-center \  // <1>
...
+    -n control-center --create-namespace \ # (2)
+    --set serviceAccount.clusterAdmin=true \  # (3)
+    -f values-ingress.yaml \ # (4)
+    --wait # (5)
+----
+
+<1> This installs Control Center.
+<2> This creates a dedicated namespace to isolate Control Center from other applications.
+<3> Grants administrative privileges that allow Control Center to manage cluster resources.
+<4> Configures a custom ingress configuration (see  below)
```
Comment (MatthewVaadin) [inline articles/control-center/getting-started/index.adoc:57]:
```suggestion
<4> Configures a custom ingress configuration (see below)
```

**Review by MatthewVaadin — CHANGES_REQUESTED** [review]
Address Vale comments

**Review by MatthewVaadin — APPROVED** [review]

File: articles/control-center/getting-started/index.adoc [hunk 2]
```diff
@@ -37,8 +39,48 @@ helm install control-center oci://docker.io/vaadin/control-center \  // <1>
...
+  className: "nginx"
+  hosts:
+    - host: "control-center.example.com"
+      paths:
+        - path: "/"
+          pathType: Prefix
+  tls:
+    - hosts:
+        - "control-center.example.com"
+      secretName: "control-center-tls"
```
Comment (jojule) [inline articles/control-center/getting-started/index.adoc:80]: If I do have such a certificate, how do I share the secret? (lets document that)

File: articles/control-center/getting-started/index.adoc [hunk 3]
```diff
@@ -37,8 +39,48 @@ helm install control-center oci://docker.io/vaadin/control-center \  // <1>
...
+----
+ingress:
+  enabled: true
+  className: "nginx"
+  hosts:
+    - host: "control-center.example.com"
+      paths:
+        - path: "/"
+          pathType: Prefix
+  tls:
```
Comment (jojule) [inline articles/control-center/getting-started/index.adoc:77]: I likely do not have a certificate yet. Thus I should probably redirect local port to communicate securely with control center at installation phase? Documentation needed.

File: articles/control-center/getting-started/index.adoc [hunk 4]
```diff
@@ -37,8 +39,48 @@ helm install control-center oci://docker.io/vaadin/control-center \  // <1>
 <4> Configures service settings for a load balancer on port `8000`. Change the port if necessary for your environment.
 <5> This says for Helm to wait until Control Center is fully deployed.
 
+=== Production environment
+
+.Terminal
+[source,bash]
+----
+helm install control-center oci://docker.io/vaadin/control-center \ # (1)
```
Comment (jojule) [inline articles/control-center/getting-started/index.adoc:47]: Should I `--set service.type=LoadBalancer` to get cluster assigning an IP for it?

**Review by jojule — CHANGES_REQUESTED** [review]
Documentation seems incomplete (see comments), but even addressing these there is a related [blocker](https://github.com/vaadin/control-center/issues/645)

File: articles/control-center/getting-started/index.adoc (hunk 4 above)
Comment (heruan) [reply articles/control-center/getting-started/index.adoc:47]: For production you should prefer to create a `Ingress` where you can specify also a certificate. Most cloud providers allow to reserve an IP address on before hand for an Ingress, e.g. in Google Cloud you add this annotation to use a reserved IP address:
```
kubernetes.io/ingress.global-static-ip-name: "my-static-ip"
```

File: articles/control-center/getting-started/index.adoc (hunk 3 above)
Comment (heruan) [reply articles/control-center/getting-started/index.adoc:77]: If you're deploying to production, you probably have a certificate already. If you're not deploying to production, you can create one with a tool like `mkcert`:
```
mkcert control-center.example.com
```
Let's document this.

File: articles/control-center/getting-started/index.adoc (hunk 2 above)
Comment (heruan) [reply articles/control-center/getting-started/index.adoc:80]: Agree, let's document how to create a secret for the certificate:
```
kubectl create secret tls control-center-tls --cert=cert.pem --key=key.pem
```

**Comment by heruan** [comment]
> Documentation seems incomplete (see comments), but even addressing these there is a related [blocker](https://github.com/vaadin/control-center/issues/645)

This happens only if you install with a port forward, should be addressed at some point by I wouldn't consider it a blocker.

File: articles/control-center/getting-started/index.adoc (hunk 3 above)
Comment (tamasmak) [reply articles/control-center/getting-started/index.adoc:102]: I documented how to create a certificate.

**Pull request merged**